samples/subsys/usb/cdc_acm/src/main.c

```diff
@@ -41,12 +41,27 @@ static void interrupt_handler(struct device *dev)
 
 static void write_data(struct device *dev, const char *buf, int len)
 {
+	int sent;
+	u32_t transfer_len;
 	uart_irq_tx_enable(dev);
 
-	data_transmitted = false;
-	uart_fifo_fill(dev, (const u8_t *)buf, len);
-	while (data_transmitted == false)
-		;
+	while (len) {
+		data_transmitted = false;
+		/* FIXME: An issue observed when USB TX Endpoint is written
+		 * more than once (greater than 4 Bytes of data is written)
+		 * (Jira 2074). So restricting writing 4 bytes at a time into
+		 * USB TX Endpoing.
+		 */
+		transfer_len = len > sizeof(u32_t) ? sizeof(u32_t) : len;
+		sent = uart_fifo_fill(dev, (const u8_t *)buf, transfer_len);
```

Comment:
Other than that fixing this code to correctly handle the return value of uart_fifo_fill() is a great idea *independent* of any bug fixing in cdc_acm_fifo_fill(). The original code ignored the return value of a write()-a-like function and that is always bogus!